microhapulator/dist.py

```diff
@@ -9,27 +9,32 @@
 
 import json
 import microhapulator
-from microhapulator.genotype import ObservedGenotype
+from microhapulator.genotype import ObservedGenotype, SimulatedGenotype
 
 
 def dist(gt1, gt2):
-    allloci = set(gt1.alleles()).union(gt2.alleles())
+    allloci = set(gt1.loci()).union(gt2.loci())
     hammdist = 0
     for locus in allloci:
-        allele1, allele2 = None, None
-        if locus in gt1.data:
-            allele1 = gt1.data[locus]['genotype']
-        if locus in gt2.data:
-            allele2 = gt2.data[locus]['genotype']
+        allele1 = gt1.alleles(locus)
+        allele2 = gt2.alleles(locus)
```

Comment:
Updated the `SimulatedGenotype` and `ObservedGenotype` classes so that their `alleles()` and `loci()` methods perform identically to take advantage of duck typing.